rats-processors/src/python/rats/pipelines_app/_decorators.py

```diff
@@ -0,0 +1,61 @@
+from collections.abc import Callable
+from typing import Concatenate, ParamSpec, TypeVar
+
+from rats import apps
+from rats.processors import ux
+
+T_ServiceType = TypeVar("T_ServiceType")
+P = ParamSpec("P")
+T_Container = TypeVar("T_Container", bound=apps.Container)
+
+
+def _process_method_to_task_method(
+    process_method: Callable[Concatenate[T_Container, P], ux.T_Processor_Output],
+) -> Callable[[T_Container], ux.UPipeline]:
+    """Convert a process method to a method returning a task wrapping over the process method."""
+
+    class _Processor(ux.IProcess):
+        process = process_method
+
+    _Processor.__name__ = process_method.__name__
+
+    def get_task(self: T_Container) -> ux.UPipeline:
+        return ux.UTask(_Processor)
+
+    get_task.__name__ = process_method.__name__
+
+    return get_task
+
+
+def task(
+    process_method: Callable[Concatenate[T_Container, P], ux.T_Processor_Output],
+) -> Callable[[T_Container], ux.UPipeline]:
+    """Decorator creating a pipeline service from a process method.
+
+    The pipeline will be a task wrapping over the method.
+
+    The name of the pipeline will be name of the method.
+    """
+    task_method = _process_method_to_task_method(process_method)
+    return apps.autoid_service(task_method)
+
+
+def pipeline(
+    get_pipeline_method: Callable[[T_Container], ux.Pipeline[ux.TInputs, ux.TOutputs]],
+) -> Callable[[T_Container], ux.Pipeline[ux.TInputs, ux.TOutputs]]:
+    """Decorator creating a pipeline service.
+
+    The name of the pipeline will be name of the method.
+    """
+    name = get_pipeline_method.__name__
+
+    def get_pipeline_and_rename(
+        self: T_Container,
+    ) -> ux.Pipeline[ux.TInputs, ux.TOutputs]:
+        pipeline = get_pipeline_method(self)
+        if pipeline.name != name:
+            pipeline = pipeline.decorate(name)
```

Comment:
This ensures pipelines within a container get the name of the method that creates them, and thus get unique names.
Pipeline authors provide names by naming the (decorated) methods.
If method pipelineC combines the pipelines from methods pipelineA and pipelineB it is guaranteed that there will be no name conflicts.

Reply:
I don't think that is the responsibility of the decorator. We can have many ways of creating a pipeline I may not want a magic decorator change the name of my pipeline. This is hidden magic, I would prefer not to do hidden things for users.

Reply:
Switch your perspective to the usage, i.e. to how pipelines are created in ExamplePipelineContainer1 and ExamplePipelineContainer2, and you'll see that the user just does not need to specify pipeline names and is guaranteed uniqueness.  Whether we do that in the decorator or in other places does not matter.

Reply:
The usage is ok, but I don't like to start accumulating decorating strings in pipelines. I think they become illegible, and I would prefer we didn't go that road, because the experience with all the wrappers that we have at the moment is not good.

I want advocate we add the option to combine pipelines without a name. Then I would be happy if this decorator decorates it's method name, but we don't decorate twice. I would avoid adding uuid hexes and then redecorating them on top.